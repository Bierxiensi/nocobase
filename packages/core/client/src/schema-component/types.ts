import type { Form } from '@formily/core';
import type { IRecursionFieldProps, ISchemaFieldProps, SchemaReactComponents } from '@formily/react';
import type React from 'react';

export interface ISchemaComponentContext {
  scope?: any;
  components?: SchemaReactComponents;
  refresh?: () => void;
  reset?: () => void;
  designable?: boolean;
  setDesignable?: (value: boolean) => void;
  SchemaField?: React.FC<ISchemaFieldProps>;
}

export interface ISchemaComponentProvider {
  designable?: boolean;
  form?: Form;
  scope?: any;
  components?: SchemaReactComponents;
}

export interface IRecursionComponentProps extends IRecursionFieldProps {
  scope?: any;
  components?: SchemaReactComponents;
}

export interface ISchemaComponentOptionsProps {
  scope?: any;
  components?: SchemaReactComponents;
  inherit?: boolean;
}
