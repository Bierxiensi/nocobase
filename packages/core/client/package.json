{
  "name": "@nocobase/client",
  "version": "0.9.3-alpha.1",
  "license": "Apache-2.0",
  "main": "lib",
  "module": "es/index.js",
  "typings": "es/index.d.ts",
  "dependencies": {
    "@antv/g2plot": "^2.4.31",
    "@dnd-kit/core": "^5.0.3",
    "@dnd-kit/sortable": "^6.0.1",
    "@emotion/css": "^11.11.0",
    "@formily/antd": "2.0.20",
    "@formily/core": "2.0.20",
    "@formily/react": "2.0.20",
    "@nocobase/evaluators": "0.9.3-alpha.1",
    "@nocobase/sdk": "0.9.3-alpha.1",
    "@nocobase/utils": "0.9.3-alpha.1",
    "ahooks": "^3.7.7",
    "antd": "^4.24.10",
    "axios": "^0.26.1",
    "classnames": "^2.3.2",
    "cron-parser": "^4.8.1",
    "cronstrue": "^2.27.0",
    "file-saver": "^2.0.5",
    "i18next": "^22.4.15",
    "i18next-http-backend": "^2.2.0",
    "markdown-it": "13.0.1",
    "markdown-it-highlightjs": "3.3.1",
    "mathjs": "^10.6.4",
    "mermaid": "9.4.3",
    "react-beautiful-dnd": "^13.1.1",
    "react-big-calendar": "^0.38.9",
    "react-drag-listview": "^0.1.9",
    "react-error-boundary": "^4.0.4",
    "react-helmet": "^6.1.0",
    "react-hotkeys-hook": "^3.4.7",
    "react-i18next": "^11.18.6",
    "react-iframe": "~1.8.5",
    "react-image-lightbox": "^5.1.4",
    "react-js-cron": "^3.2.0",
    "react-quill": "^1.3.5",
    "react-router-dom": "^5.3.4",
    "react-to-print": "^2.14.12",
    "sanitize-html": "2.10.0",
    "solarlunar-es": "^1.0.9",
    "use-deep-compare-effect": "^1.8.1"
  },
  "peerDependencies": {
    "@types/react": ">=16.8.0 || >=17.0.0",
    "@types/react-dom": ">=16.8.0 || >=17.0.0",
    "react": ">=16.8.0 || >=17.0.0",
    "react-dom": ">=16.8.0",
    "react-is": ">=16.8.0 || >=17.0.0"
  },
  "devDependencies": {
    "@testing-library/react": "^12.1.5",
    "@types/markdown-it": "12.2.3",
    "@types/markdown-it-highlightjs": "3.3.1",
    "axios-mock-adapter": "^1.21.4"
  },
  "gitHead": "ce588eefb0bfc50f7d5bbee575e0b5e843bf6644"
}
