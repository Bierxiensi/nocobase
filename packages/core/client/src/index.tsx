import 'antd/dist/antd.css';
import './global.less';
export * as reactI18nNext from 'react-i18next';
export * as antd from 'antd';
export * as formilyReact from '@formily/react';
export * as antdIcons from '@ant-design/icons';
export * as emotionCss from '@emotion/css';

export * from './acl';
export * from './antd-config-provider';
export * from './api-client';
export * from './application';
export * from './async-data-provider';
export * from './block-provider';
export * from './board';
export * from './china-region';
export * from './collection-manager';
export * from './document-title';
export * from './i18n';
export * from './icon';
export * from './plugin-manager';
export * from './pm';
export * from './powered-by';
export * from './record-provider';
export * from './route-switch';
export * from './schema-component';
export * from './schema-initializer';
export * from './schema-settings';
export * from './schema-templates';
export * from './settings-form';
export * from './system-settings';
export * from './user';
