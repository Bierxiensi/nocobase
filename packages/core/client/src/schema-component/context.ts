import { createContext } from 'react';
import type { ISchemaComponentContext } from './types';

export const SchemaComponentContext = createContext<ISchemaComponentContext>({});
