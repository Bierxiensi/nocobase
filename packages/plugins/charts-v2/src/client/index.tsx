import {
  SchemaComponentOptions,
  SchemaInitializerContext,
  SchemaInitializerProvider,
  ACLCollectionProvider,
} from '@nocobase/client';
import React, { useContext } from 'react';
import { ChartV2BlockInitializer, ChartV2BlockDesigner, ChartInitializers, ChartV2Block } from './block';
import { ChartRenderer, G2PlotLibrary, ChartRendererProvider } from './renderer';
import { ChartLibraryProvider } from './renderer/ChartLibrary';

const Chart: React.FC = (props) => {
  const initializers = useContext(SchemaInitializerContext);
  const children = initializers.BlockInitializers.items[0].children;
  const has = children.some((initializer) => initializer.title === 'ChartV2');
  if (!has) {
    children.push({
      key: 'chart-v2',
      type: 'item',
      title: 'ChartV2',
      component: 'ChartV2BlockInitializer',
    });
  }
  return (
    <SchemaComponentOptions
      components={{ ChartV2BlockInitializer, ChartRenderer, ChartV2BlockDesigner, ChartV2Block, ChartRendererProvider }}
    >
      <SchemaInitializerProvider initializers={{ ...initializers, ChartInitializers }}>
        <ChartLibraryProvider name="G2Plot" charts={G2PlotLibrary}>
          {props.children}
        </ChartLibraryProvider>
      </SchemaInitializerProvider>
    </SchemaComponentOptions>
  );
};

export default Chart;
