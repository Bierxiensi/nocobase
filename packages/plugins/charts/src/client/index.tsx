import { registerValidateRules } from '@formily/core';
import { SchemaComponentOptions, SchemaInitializerContext, SettingsCenterProvider } from '@nocobase/client';
import JSON5 from 'json5';
import React, { useContext } from 'react';
import { ChartBlockEngine } from './ChartBlockEngine';
import { ChartBlockInitializer } from './ChartBlockInitializer';
import { ChartQueryMetadataProvider } from './ChartQueryMetadataProvider';
import './Icons';
import { lang } from './locale';
import { CustomSelect } from './select';
import { QueriesTable } from './settings/QueriesTable';

registerValidateRules({
  json5: (value, rule) => {
    if (!value) {
      return '';
    }
    try {
      const val = JSON5.parse(value);
      if (!isNaN(val)) {
        return {
          type: 'error',
          message: lang('Invalid JSON format'),
        };
      }
      return '';
    } catch (error) {
      console.error(error);
      return {
        type: 'error',
        message: lang('Invalid JSON format'),
      };
    }
  },
});

export default React.memo((props) => {
  const items = useContext(SchemaInitializerContext);
  const children = items.BlockInitializers.items[0].children;
  children.push({
    key: 'chart',
    type: 'item',
    icon: 'PieChartOutlined',
    title: '{{t("Chart",{ns:"charts"})}}',
    component: 'ChartBlockInitializer',
  });
  return (
    <ChartQueryMetadataProvider>
      <SettingsCenterProvider
        settings={{
          charts: {
            title: '{{t("Chart",{ns:"charts"})}}',
            icon: 'PieChartOutlined',
            tabs: {
              queries: {
                title: '{{t("Queries",{ns:"charts"})}}',
                component: () => <QueriesTable />,
              },
            },
          },
        }}
      >
        <SchemaComponentOptions components={{ CustomSelect, ChartBlockInitializer, ChartBlockEngine }}>
          <SchemaInitializerContext.Provider value={items}>{props.children}</SchemaInitializerContext.Provider>
        </SchemaComponentOptions>
      </SettingsCenterProvider>
    </ChartQueryMetadataProvider>
  );
});
