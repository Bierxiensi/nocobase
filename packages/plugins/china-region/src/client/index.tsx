import { Plugin } from '@nocobase/client';
import { ChinaRegionProvider } from './ChinaRegionProvider';

export class ChinaRegionPlugin extends Plugin {
  async load() {
    this.app.use(ChinaRegionProvider);
  }
}

export default ChinaRegionPlugin;
