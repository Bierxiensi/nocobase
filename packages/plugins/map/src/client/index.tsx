import { Plugin } from '@nocobase/client';
import { MapProvider } from './MapProvider';

export class MapPlugin extends Plugin {
  async load() {
    this.app.use(MapProvider);
  }
}

export default MapPlugin;
