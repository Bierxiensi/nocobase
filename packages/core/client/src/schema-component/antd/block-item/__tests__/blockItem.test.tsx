import { render, screen, fireEvent } from '@testing-library/react';
import { observer, useFieldSchema } from '@formily/react';
import { uid } from '@formily/shared';
import { DndContext, DragHandler } from '@nocobase/client';
import React from 'react';
import { BlockItem } from '../BlockItem';
import { SchemaComponent, SchemaComponentProvider } from '../../../../schema-component';

const Block = observer((props) => {
  const fieldSchema = useFieldSchema();
  return (
    <div style={{ marginBottom: 20, padding: '0 20px', height: 50, lineHeight: '50px', background: '#f1f1f1' }}>
      Block {fieldSchema.name}
      <DragHandler />
    </div>
  );
});

const App = () => {
  return (
    <SchemaComponentProvider components={{ DndContext, BlockItem, Block }}>
      <SchemaComponent
        schema={{
          type: 'void',
          name: 'page',
          'x-component': 'DndContext',
          'x-uid': uid(),
          properties: {
            block1: {
              'x-decorator': 'BlockItem',
              'x-component': 'Block',
              'x-uid': uid(),
            },
            block2: {
              'x-decorator': 'BlockItem',
              'x-component': 'Block',
              'x-uid': uid(),
            },
            block3: {
              'x-decorator': 'BlockItem',
              'x-component': 'Block',
              'x-uid': uid(),
            },
          },
        }}
      />
    </SchemaComponentProvider>
  );
};

describe('blockItem', () => {
  it('blockItem', () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
});
