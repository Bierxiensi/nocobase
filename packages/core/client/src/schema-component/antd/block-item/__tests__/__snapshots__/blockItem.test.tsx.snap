// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`blockItem blockItem 1`] = `
<div>
  <div
    class="nb-block-item css-r8u0a4"
  >
    <div
      style="margin-bottom: 20px; padding: 0px 20px; height: 50px; line-height: 50px; background: rgb(241, 241, 241);"
    >
      Block 
      block1
      <div
        style="display: inline-block; width: 12px; height: 12px; line-height: 12px; text-align: left;"
      >
        <div
          aria-describedby="DndDescribedBy-0"
          aria-roledescription="draggable"
          role="button"
          style="position: relative; z-index: 1; line-height: 0; height: 2px; width: 2px; font-size: 0; display: inline-block;"
          tabindex="0"
        >
          <span
            style="cursor: move; font-size: 14px;"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="nb-block-item css-r8u0a4"
  >
    <div
      style="margin-bottom: 20px; padding: 0px 20px; height: 50px; line-height: 50px; background: rgb(241, 241, 241);"
    >
      Block 
      block2
      <div
        style="display: inline-block; width: 12px; height: 12px; line-height: 12px; text-align: left;"
      >
        <div
          aria-describedby="DndDescribedBy-0"
          aria-roledescription="draggable"
          role="button"
          style="position: relative; z-index: 1; line-height: 0; height: 2px; width: 2px; font-size: 0; display: inline-block;"
          tabindex="0"
        >
          <span
            style="cursor: move; font-size: 14px;"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="nb-block-item css-r8u0a4"
  >
    <div
      style="margin-bottom: 20px; padding: 0px 20px; height: 50px; line-height: 50px; background: rgb(241, 241, 241);"
    >
      Block 
      block3
      <div
        style="display: inline-block; width: 12px; height: 12px; line-height: 12px; text-align: left;"
      >
        <div
          aria-describedby="DndDescribedBy-0"
          aria-roledescription="draggable"
          role="button"
          style="position: relative; z-index: 1; line-height: 0; height: 2px; width: 2px; font-size: 0; display: inline-block;"
          tabindex="0"
        >
          <span
            style="cursor: move; font-size: 14px;"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
